### PR TITLE
ci: ci, cd, various cleanup

### PR DESCRIPTION
**Pull request by cristianmtr**

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+scipy==1.6.1
```
Comment (hanxiao) [inline requirements.txt:1]: not required

File: .github/workflows/scripts/get-all-test-paths.sh [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  # only if the folder has a tests or a Dockerfile but excluding integration tests (always run & separate)
+  if [[ -f "${file_base_dir}/Dockerfile" || -d "${file_base_dir}/tests/" ]]; then
+    if [[ ! " ${changed_folders[@]} " =~ " ${file_base_dir} " ]]; then
+        changed_folders+=(${file_base_dir})
+    fi
+  fi
+done
+
+output=$(jq --compact-output --null-input '$ARGS.positional' --args "${changed_folders[@]}")
+echo "::set-output name=matrix::${output}"
```
Comment (hanxiao) [inline .github/workflows/scripts/get-all-test-paths.sh:28]: unnecessary file

File: .github/workflows/scripts/get-gpu-executors.sh [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+        gpu_folders+=(${file_base_dir})
+      fi
+    fi
+  fi
+  cd $root
+done
+
+#echo will store gpu_folders in output matrix
+output=$(jq --compact-output --null-input '$ARGS.positional' --args "${gpu_folders[@]}")
+echo "::set-output name=matrix::${output}"
```
Comment (hanxiao) [inline .github/workflows/scripts/get-gpu-executors.sh:35]: unnecessary file

File: .github/workflows/scripts/push_gpu.sh [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+curl -H "Authorization: token $token" -H 'Accept: application/vnd.github.v3.raw' -O https://api.github.com/repos/jina-ai/executors-secrets/contents/secrets.json
+
+exec_uuid=`cat secrets.json | jq -r '.[] | select(.Alias=="'$exec_name'").UUID8'`
+echo "::add-mask::$exec_uuid"
+echo UUID=`head -c 3 <(echo $exec_uuid)`
+
+exec_secret=`cat secrets.json | jq -r '.[] | select(.Alias=="'$exec_name'").Secret'`
+echo "::add-mask::$exec_secret"
+echo SECRET=`head -c 3 <(echo $exec_secret)`
+
```
Comment (hanxiao) [inline .github/workflows/scripts/push_gpu.sh:33]: unnecessary file

File: .github/workflows/scripts/test_docker.sh [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+  fi
+  echo ~~~~~~~OUTPUT BELOW~~~~~~~
+  cat nohup.out
+else
+  echo no Dockerfile, nothing to test
+  local_exit_code=0
+fi
+
+echo final exit code = $local_exit_code
+exit $local_exit_code
```
Comment (hanxiao) [inline .github/workflows/scripts/test_docker.sh:53]: unnecessary file

**Review by hanxiao — DISMISSED** [review]
please delete all unnecessary file

File: .github/workflows/cd.yml [hunk 6]
```diff
@@ -0,0 +1,41 @@
+name: CD
+
+on:
+  push:
+    branches:
+    - $default-branch
+  workflow_dispatch:
+  pull_request:
```
Comment (hanxiao) [inline .github/workflows/cd.yml:8]: CD on pull request?

Comment (cristianmtr) [reply .github/workflows/cd.yml:8]: Testing

**Comment by cristianmtr** [comment]
The files will be necessary, in CI. I just need to adapt them to a single Executor repo

**Review by hanxiao — APPROVED** [review]

**Pull request merged**